cargo-shuttle/src/config.rs

```diff
@@ -122,17 +122,17 @@ impl ConfigManager for LocalConfigManager {
 /// Global client config for things like API keys.
 #[derive(Deserialize, Serialize, Default)]
 pub struct GlobalConfig {
-    pub api_key: Option<ApiKey>,
+    api_key: Option<String>,
     pub api_url: Option<ApiUrl>,
 }
 
 impl GlobalConfig {
-    pub fn api_key(&self) -> Option<&ApiKey> {
-        self.api_key.as_ref()
+    pub fn api_key(&self) -> Option<Result<ApiKey>> {
+        self.api_key.as_ref().map(|key| ApiKey::parse(key))
```

Comment:
I am just thinking that would've been best to avoid the parsing each time we call `api_key`, but only if not needed.

Reply:
My thinking here is that since the API-key will be stored as a string in the config file, and this can be freely modified manually, every time we read it from the config it needs to be parsed. There may well be a better way to do this though :thinking: